### PR TITLE
fix: Improve download action documentation

### PR DESCRIPTION
**Pull request by harshilp24**
- Added a note that the mime-type parameter of the file can't be forced or manipulated.
- Replaced GIFs with videos (except the main one).
- Lintr fixes.
- Added Troubleshooting callout and a few minor changes.

### CODE REVIEW

File: website/docs/reference/appsmith-framework/widget-actions/download.md [hunk 1]
```diff
@@ -3,48 +3,65 @@ sidebar_position: 6
 ---
 # Download
 
-This action is used to download any data as a file. We use [downloadjs](https://github.com/rndme/download) for this functionality
+With Appsmith, users have the ability to download a wide range of file formats. This action serves the purpose of downloading any data as a file. The feature is made possible by using  [downloadjs](https://github.com/rndme/download) library.
```
Comment (danciaclara) [inline website/docs/reference/appsmith-framework/widget-actions/download.md:6]: Hey why did we change this? Was it being flagged by linter, else let us keep it as before

File: website/docs/reference/appsmith-framework/widget-actions/download.md [hunk 2]
```diff
@@ -3,48 +3,65 @@ sidebar_position: 6
...
+* HTML
+* CSV
+* JSON
+* JPEG
+* PNG
+* SVG
 
-    To download a plain text, the _expected data_ is a string representing the text content to be downloaded.
+:::info
+It's important to note that the mime-type parameter of the file can't be forced or manipulated, and the system only allows the file to be downloaded in its original mime-type.
```
Comment (danciaclara) [inline website/docs/reference/appsmith-framework/widget-actions/download.md:35]:
```suggestion
The Download action does not convert the file to a specific type and it is downloaded in the original format. If you need to change the file type, you would need to convert the data to a specific format using JavaScript before downloading.
```

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/reference/appsmith-framework/widget-actions/download.md (hunk 1 above)
Comment (harshilp24) [reply website/docs/reference/appsmith-framework/widget-actions/download.md:6]: Yes, there were a few lintr issues, so I rephrased the introduction.

File: website/docs/reference/appsmith-framework/widget-actions/download.md [hunk 3]
```diff
@@ -3,48 +3,65 @@ sidebar_position: 6
...
 
 
-![Click to expand](/img/download-image.gif)
+### Downloading a file
 
-*   **Downloading a file**
+Files of various types can be downloaded using the download action by providing the URL of the file to be downloaded.
 
-    By supplying the URL of the file to be downloaded, files of different types can be downloaded via the download action.
+> For files to be successfully downloaded, **their contents must be served over HTTPS** to prevent requests from being blocked. To prevent Cross-Origin Resource Sharing (CORS) errors, ensure that the server where the file is fetched from is **CORS-enabled** and returns the required headers in the response.
```
[the remark this answers is not in the document]
Comment (harshilp24) [reply website/docs/reference/appsmith-framework/widget-actions/download.md:64]: DONE

**Comment by bharath31** [comment]
@harshilp24 could you also add code samples for downloading files from a URL? this is a common use case. 

Sample code for downloading a PDF file from an API response  
```
downloadPDF: async () => {
	let data = getPdf.data
	const blob = new Blob([data], {type: 'application/pdf'});
	const url = URL.createObjectURL(blob);
	await download(url, "sample.pdf", "application/pdf")
}
```

**Review by danciaclara — APPROVED** [review]

**Review by bharath31 — DISMISSED** [review]

**Review by danciaclara — APPROVED** [review]

**Pull request merged**